### PR TITLE
FIX: Match the doc description while copying over data

### PR DESCRIPTION
**Pull request by MridulS**
As pointed out by @eriknw in https://github.com/networkx/networkx/pull/7066#discussion_r1374017287, we weren't following our own docs :)

### CODE REVIEW

**Review by rossbar — APPROVED** [review]
Thanks @MridulS & @eriknw !

File: networkx/algorithms/operators/tests/test_binary.py [hunk 1]
```diff
@@ -199,6 +199,8 @@ def test_difference_attributes():
     assert set(gh.nodes()) == set(g.nodes())
     assert set(gh.nodes()) == set(h.nodes())
     assert sorted(gh.edges()) == []
+    # node data should not be copied over
+    assert gh.nodes.data() != g.nodes.data()
```
Comment (dschult) [inline networkx/algorithms/operators/tests/test_binary.py:203]: Also for graph attributes...

**Comment by MridulS** [comment]
> Does this need a deprecation? I'm fine with either way.

We are fixing a bug, so I would lean on no.

**Review by dschult — APPROVED** [review]
Yay -- Thanks for this!

**Pull request merged**